python/rmm/_lib/device_buffer.pyx

```diff
@@ -250,15 +250,14 @@ cdef class DeviceBuffer:
 
         cdef device_buffer* dbp = self.c_obj.get()
 
-        with nogil:
```

Comment:
Why did we remove the `nogil` here?

Reply:
copy_device_to_ptr now is a gil function, and goes nogil internally, due to the CudaStreamView parameter. Maybe I'm doing this wrong.